testing/e2e/git_rules_testkit.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+
+# Title        git_rules_testkit.sh
+# Description  Test kit for glueing together Bazel wiring and git_cached repo rules
+#==================================================================================
+
+source ./testing/e2e/bazel.sh
+
+function recreate_git_cached_rule_for_small_repo() {
+  # Re-create WORKSPACE file and re-declare git_cached_repository with the new commit hash
+  create_workspace_file_targeting_wix_build_tools "${WORKSPACE_DIR}/WORKSPACE"
+  local commit_hash=$1
+  local cache_dir=${2:-"default-dir"}
+  create_git_cached_rule_for_small_repo ${commit_hash} ${cache_dir}
+}
+
+function create_git_cached_rule_for_small_repo() {
+  local commit_hash=$1
+  local cache_dir=${2:-"default-dir"}
+
+  # Trim consecutive // characters since Bazel would fail if such exists on a target path
+  local trimmed_path=$(echo ${WORKSPACE_DIR} | tr -s '/')
+
+  if [[ ${cache_dir} == "default-dir" ]]; then
+    cat >>${WORKSPACE_DIR}/WORKSPACE <<EOF
+load("@wix_build_tools//rules/git:git_cached_repository.bzl", "git_cached_repository")
+git_cached_repository(
+    name = "small_repo",
+    remote_url = "${REPO_SMALL_DIR}",
+    commit = "${commit_hash}",
+    branch = "master",
+)
+EOF
+  else
+    cat >>${WORKSPACE_DIR}/WORKSPACE <<EOF
+load("@wix_build_tools//rules/git:git_cached_repository.bzl", "git_cached_repository")
+git_cached_repository(
+    name = "small_repo",
+    remote_url = "${REPO_SMALL_DIR}",
+    commit = "${commit_hash}",
+    branch = "master",
+    cache_directory = "${cache_dir}",
```

Comment:
Maybe there's a way for you to DRY here? 
Write the HEADER
```python
load("@wix_build_tools//rules/git:git_cached_repository.bzl", "git_cached_repository")
git_cached_repository(
    name = "small_repo",
    remote_url = "${REPO_SMALL_DIR}",
    commit = "${commit_hash}",
    branch = "master",
```

Then using `if` statememnt write/not-write the line:
```python
    cache_directory = "${cache_dir}",
```
then write the closing?

Reply:
What about the next attribute that we would like to have like `branch`, `shallow_since` etc..?
I'll see how to create a dynamic behavior of accepting different function arguments and then assigning the relevant attribute to the `git_cached_repository` similar to that you've suggested.